createtransfers/createtransfers.py

```diff
@@ -0,0 +1,366 @@
+
+    rm_dirs_and_create(new_media_dir)
+    copy_media_location(media_path, new_media_dir, media_n, media_ext)
+
+
+def copy_media_location(copy_media_path, new_media_dir, media_n, media_ext):
+    for i in range(media_n):
+        new_media_fname = 'mediafile_{0}.{1}'.format(i, media_ext)
+        new_media_path = os.path.join(new_media_dir, new_media_fname)
+        shutil.copyfile(copy_media_path, new_media_path)
+    logger.info("%s files created at %s", media_n, new_media_dir)
+
+
+def stat_file(file_path, n):
+    """ Check that the file exists to make sure we're not referencing objects
+    elsewhere in the transfer structure that don't exist. Return something
+    useful if the files are available.
+    """
+    if not os.path.exists(file_path):
+        logger.error("Cannot find media file at location %s", file_path)
+        return ERR_OPERATION_FAILED
+    logger.info(("%s duplicated %s times "
+                 "will take up %sGB disk space"), file_path, n,
+                ((os.path.getsize(file_path)
+                 * n) / 1024.0) / 1024.0 / 1024.0)
+
+
+def create_variously_encoded_files(**kwargs):
+    """Create files with strange (non-UTF8) encodings under
+    TestTransfers/files_with_various_encodings/
+    """
+    target_path = os.path.join(
+        HERE, 'TestTransfers', 'files_with_various_encodings')
+
+    logger.info("Transfer target path: %s", target_path)
+
+    for encoding_info in VARIOUS_ENCODINGS:
+        encoding_dir_path = os.path.join(
+            target_path, encoding_info['dir_name'])
+        success = rm_dirs_and_create(encoding_dir_path)
+        if success is not None:
+            return success
+        encoding = encoding_info['encoding']
+        if isinstance(encoding_info['file_name'], tuple):
+            for fname in encoding_info['file_name']:
+                success = create_file_and_write(fname,
+                                                encoding_dir_path,
+                                                encoding)
+                if success is not None:
+                    return success
+        else:
+            success = create_file_and_write(encoding_info['file_name'],
+                                            encoding_dir_path,
+                                            encoding)
+            if success is not None:
+                return success
+
+
+def create_large_test_transfers(**kwargs):
+    """Create large test transfers by copying existing sampledata files
+    multiple times to specific subdirectories of
+    TestTransfers/acceptance-tests/performance/
+    """
+    image_n = 113
+    video_n = 669
+
+    performance_path = os.path.join(
+        HERE, 'TestTransfers', 'acceptance-tests', 'performance')
+    rm_dirs_and_create(performance_path)
+
+    # Create dir images-17M-each-2G-total/ containing 2G of 17M image files.
+    image_path = os.path.join(
+        HERE, 'TestTransfers', 'manualNormalization', 'manualNormalization',
```

Comment:
@jrwdunham can you double clean your sampledata repository here and see if the folder still exists. In my working copy there wasn't such a folder so this change was to make the generation of this set work.

Reply:
Okay, I think I see... that is the path, but the file you previously had it pointed at `image_8.tif` isn't there:

https://github.com/artefactual/archivematica-sampledata/tree/master/TestTransfers/ManualNormalization/manualNormalization/preservation

It's here:

https://github.com/artefactual/archivematica-sampledata/blob/master/TestTransfers/manualNormalization/manualNormalization/preservation/image_8.tif

We have two folders with confusingly similar names! 

(I personally would like to overhaul this collection to make it a little more self-describing and give each folder a README.md. Not that it would have helped here.